src/spikeinterface/preprocessing/normalize_scale.py

```diff
@@ -17,14 +17,15 @@ def __init__(self, parent_recording_segment, gain, offset, dtype):
         self.offset = offset
         self._dtype = dtype
 
-    def get_traces(self, start_frame, end_frame, channel_indices):
+    def get_traces(self, start_frame, end_frame, channel_indices) -> np.ndarray:
         traces = self.parent_recording_segment.get_traces(start_frame, end_frame, channel_indices)
-        scaled_traces = traces * self.gain[:, channel_indices] + self.offset[:, channel_indices]
+        traces = np.multiply(traces, self.gain[:, channel_indices], out=traces)
```

Comment:
@alejoe91 @samuelgarcia  note that the following will also work in the same way and avoid extra memory allocations:
```python      
      traces *= self.gain[:, channel_indices]
      traces += self.offset[:, channel_indices]
```

If you prefer how that looks but we will need to make a comment like `#Do not move this` whereas the `out` argument makes it more clear (in my view) what is going on. But I don't feel strong about this.

Your call guys

Reply:
Ah forget about it. I actually need the full call to keep backwards compatability with usage.